pkg/kubernetes/scripts/virtual-machines/components/createVmDialog.jsx

```diff
@@ -0,0 +1,137 @@
+/*jshint esversion: 6 */
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2017 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import cockpit from 'cockpit';
+import React, { PropTypes } from "react";
+
+const _ = cockpit.gettext;
+
+import './createVmDialog.less';
+
+
+/* Create a virtual machine
+ * props:
+ *  - valuesChanged callback for changed values with the signature (key, value)
+ */
+class CreateVmDialog extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            resource: '',
+            resourceError: null,
+        };
+
+        this.getResult = this.getResult.bind(this);
+        this.validate = this.validate.bind(this);
+
+        if (this.props.valuesChanged) {
+            this.props.valuesChanged('getResult', this.getResult);
+            this.props.valuesChanged('validate', this.validate);
+        }
+    }
+
+    onChangedEventValue(key, e) {
+        if (e && e.target && typeof(e.target.value) !== 'undefined') {
+            this.onChangedValue(key, e.target.value);
+        }
+    }
+
+    onChangedValue(key, value) {
```

Comment:
- To prepare for additional values in the future. 
-  To wrap and not duplicate event handlers (`onChangedEventValue`)

Example of that in machines create vm dialog:
 - https://github.com/suomiy/cockpit/blob/virt-create/pkg/machines/components/createVmDialog.jsx#L132

Reply:
IMO, this decreases readability while not bringing clear benefit. 
Anyway it works, so I would not block merge on that if case I'm the only-one willing to avoid it. @jniederm ?